src/js/gi/selectors/estimator.js

```diff
@@ -0,0 +1,203 @@
+import { createSelector } from 'reselect';
+
+const getConstants = (state) => state.constants.constants;
+
+const getEligibilityDetails = (state) => {
+  const details = Object.assign({}, state.eligibility);
+  delete details.dropdowns;
+  return details;
+};
+
+const getRequiredAttributes = (_state, props) => {
+  const { type, bah } = props;
+  return { type, bah };
+};
+
+function getDerivedAttributes(constant, eligibility, institution) {
+  const your = eligibility;
+  const its = institution;
+  let serviceDischarge, vre911Eligible, tier, oldGiBill, onlyVRE, onlyTuitionFees, monthlyRate;
+
+  serviceDischarge = (your.cumulative_service === 'service discharge');
+
+  // VRE and post-9/11 eligibility
+  vre911Eligible = (your.gi_bill_chapter === '31' && your.elig_for_post_gi_bill === 'yes');
+
+  // VRE-without-post-911 eligibility
+  onlyVRE = (your.gi_bill_chapter === '31' && your.elig_for_post_gi_bill === 'no');
+
+  // Determines benefits tier
+  tier = (vre911Eligible || serviceDischarge) ? 1 : Number(your.cumulative_service);
+
+  oldGiBill = (
+    your.gi_bill_chapter === '30'
+    || your.gi_bill_chapter === '1607'
+    || your.gi_bill_chapter === '1606'
+    || your.gi_bill_chapter === '35'
+  );
+
+  // Determines whether monthly benefit can only be spent on tuition/fees
+  const activeDutyThirtyOr1607 = (
+    your.military_status === 'active duty' &&
+    (your.gi_bill_chapter === '30' || your.gi_bill_chapter === '1607')
+  );
+  const correspondenceOrFlightUnderOldGiBill = (
+    (its.type === 'correspondence' || its.type === 'flight') && oldGiBill === true
+  );
+  if (activeDutyThirtyOr1607 || correspondenceOrFlightUnderOldGiBill) {
+    onlyTuitionFees = true;
+  } else {
+    onlyTuitionFees = false;
+  }
+
+  // The monthly benefit rate for non-chapter 33 benefits
+  const isOJT = its.type === 'ojt';
+  const isFlight = its.type === 'flight';
+  const n = Number(your.number_of_dependents);
```

Comment:
Here's a neat little shorthand for Number conversions:

`const n = +your.number_of_dependents`.